archinstall/lib/user_interaction.py

```diff
@@ -939,3 +939,83 @@ def select_locale_enc(default):
 	).run()
 
 	return selected_locale
+
+def generic_select(p_options :Union[list,dict],
+				input_text :str = "Select one of the values shown below: ",
+				allow_empty_input :bool = True,
+				options_output :bool = True,   # function not available
+				sort :bool = False,
+				multi :bool = False,
+				default :Any = None) -> Any:
+	from copy import copy
+	"""
+	A generic select function that does not output anything
+	other than the options and their indexes. As an example:
+
+	generic_select(["first", "second", "third option"])
+		first
+		second
+		third option
+	When the user has entered the option correctly,
+	this function returns an item from list, a string, or None
+
+	Default value must be in the list of options
+	sort will be handled by Menu()
+	"""
+	# I (shallow) copy the options object. This is an input only parameter
+	options = copy(p_options)
+	# Checking if the options are different from `list` or `dict` or if they are empty
+	# duplicate of menu, but i need to check it prior as i do some type dependent processing
+	if not isinstance(options, (list,tuple, dict)):
```

Comment:
Since we moved these checks now into the menu class I think we can savely remove them from here now :)

Reply:
I still need to check, in the module, if the options is an iterable, to be able to do the map processing further on. And the copy ... Might be if I do 
```python
if type(p_options) == dict:
    options =list(p_options.values)
else:
    options = list(p_options)
```
I solve almost everything. Well, if it isn't an iterable it will crash, but we can live with it. It should only happen during development time.
Have to do a couple of tests